inputremapper/injection/macros/tasks/toggle.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# input-remapper - GUI for device specific keyboard mappings
+# Copyright (C) 2025 sezanzeb <b8x45ygc9@mozmail.com>
+#
+# This file is part of input-remapper.
+#
+# input-remapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# input-remapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with input-remapper.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import asyncio
+
+from inputremapper.injection.macros.argument import ArgumentConfig
+from inputremapper.injection.macros.macro import Macro
+from inputremapper.injection.macros.task import Task
+from inputremapper.injection.macros.tasks.wait import WaitTask
+
+
+class ToggleTask(Task):
+    """Toggle macros."""
+
+    argument_configs = [
+        ArgumentConfig(
+            name="macro",
+            position=0,
+            types=[Macro],
+        ),
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.release_count = 0
+
+    def release_trigger(self) -> None:
+        Task.release_trigger(self)
+        self.release_count += 1
+
+    async def run(self, callback) -> None:
+        macro = self.get_argument("macro").get_value()
+        macro_has_wait = any(isinstance(task, WaitTask) for task in macro.tasks)
+
+        while self.release_count <= 1:
+            await macro.run(callback)
+            if not macro_has_wait:
+                await asyncio.sleep(0.1)
```

Comment:
`hold` and `repeat` aren't doing this. And what if someone needs this to be as fast as possible.

However, `hold` does not freeze. `hold` always adds 1ms of sleep, which is already enough to keep things running smoothly.

I'd suggest we keep the behavior as consistent as possible across all repeating macros.